api/handlers_test.go

```diff
@@ -217,6 +222,85 @@ func testHandleBlockRequest(t *testing.T) {
 	}
 }
 
+// testHandleBlocklistGET verifies the GET /blocklist endpoint
+func testHandleBlocklistGET(t *testing.T) {
+	// create a context with timeout
+	ctx, cancel := context.WithTimeout(context.Background(), database.MongoDefaultTimeout)
+	defer cancel()
+
+	// create a new test API
+	skyd := &mockSkyd{}
+	api, err := newTestAPI("HandleBlockRequest", skyd)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// create a helper that fetches the blocklist
+	fetchBlocklist := func() (BlocklistGET, error) {
```

Comment:
We really need to stop this in our smaller service repos. It's just making our codebases harder to maintain over time. There should be a dedicated method on some sort of client object which already knows how to parse a json object from a response body. Then we extend that client like we do in skyd.

Reply:
I agree. I discovered there's a thing called a `ResponseRecorder` which is quite neat. You can just call the handler directly and pass a recorder. For the time being, I just mostly copy-pasted this helper to a new struct called the `APITester`, but when we extend that we can extract all duplicated code into helpers on that struct.

Does that work for you or would you've liked to see it done differently.